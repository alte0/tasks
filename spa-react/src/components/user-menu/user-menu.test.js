import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import UserMenu from './user-menu';
import {applyMiddleware, compose, createStore} from "redux";
import thunkMiddleware from "redux-thunk";
import rootReducer from "../../reducers";
import {Provider} from "react-redux";

const middlewareEnhancer = applyMiddleware(thunkMiddleware);
const composedEnhancers = compose(middlewareEnhancer);
const store = createStore(rootReducer, undefined, composedEnhancers);

it('UserMenu render', () => {
    const user = {
        name: 'Пётр',
        surname: 'Петров',
        patronymic: 'Петрович'
    }

    const tree = renderer
        .create(
            <Provider store={store}>
                <Router>
                    <UserMenu user={user} url={'/'} />
                </Router>
            </Provider>
            )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
