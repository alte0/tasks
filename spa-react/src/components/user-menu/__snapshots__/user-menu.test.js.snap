// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenu render 1`] = `
<nav
  className="user-menu"
>
  <span
    className="user-menu__user-name"
  >
      
  </span>
  <a
    className="user-menu__link link"
    href="/my-tasks-done"
    onClick={[Function]}
  >
    Выполненные задачи мною
  </a>
  <a
    className="user-menu__link link"
    href="/designated-tasks"
    onClick={[Function]}
  >
    Я назначил задачи
  </a>
  <a
    className="user-menu__link link"
    href="/add-task"
    onClick={[Function]}
  >
    Поставить задачу
  </a>
  <a
    className="user-menu__link user-menu__logout link"
    href="/log-out"
    onClick={[Function]}
  >
    Выйти
  </a>
</nav>
`;
